src/music/dmusic.cpp

```diff
@@ -832,8 +832,9 @@ static void MidiThreadProc(void *)
 			/* end? */
 			if (current_block == current_file.blocks.size()) {
 				if (current_segment.loop) {
-					current_block = 0;
+					current_block = current_segment.start_block;
 					clock->GetTime(&playback_start_time);
+					playback_start_time -= current_file.blocks[current_block].realtime * MIDITIME_TO_REFTIME;
```

Comment:
I haven't been able to figure out why yet, but the DirectMusic driver loops the title song _too_ tightly, producing a different wrong-sounding result.

Reply:
Hmm, actually, DMusic sequences all by messages by the absolute timestamp that is sent. If you subtract too much from the playback start time, the new messages might be sequenced before the old messages.

You might try setting the playback start time to `block_time`, i.e. the last transmitted timestamp and not the current time.

Reply:
Yep basing it off `block_time` fixes the issue.